src/qiskit_qec/utils/stim_tools.py

```diff
@@ -0,0 +1,347 @@
+
+"""Tools to use functionality from Stim."""
+from typing import List, Dict
+from math import log
+from stim import Circuit as StimCircuit
+from stim import DetectorErrorModel as StimDetectorErrorModel
+from stim import DemInstruction as StimDemInstruction
+from stim import DemTarget as StimDemTarget
+
+import numpy as np
+import rustworkx as rx
+
+
+from qiskit import QuantumCircuit
+from qiskit_aer.noise.errors.quantum_error import QuantumChannelInstruction
+from qiskit_aer.noise import pauli_error
+from qiskit_qec.utils.decoding_graph_attributes import DecodingGraphNode, DecodingGraphEdge
+
+
+def get_stim_circuits(circuit_dict):
+    """Converts compatible qiskit circuits to stim circuits.
+
+    Args:
+        circuit_dict: Dictionary with Qiskit circuits as values. Compatible gates are paulis,
+        controlled paulis, h, s, and sdg, swap, reset, measure and barrier. Compatible noise
+        operators correspond to a single or two qubit pauli channel.
+
+    Returns:
+        stim_circuits: TODO
+        stim_measurement_data: TODO
+    """
+    stim_circuits = {}
+    stim_measurement_data = {}
+    for circ_label, circuit in circuit_dict.items():
+        stim_circuit = StimCircuit()
+
+        # Dictionaries are not complete. For the stim definitions see:
+        #   https://github.com/quantumlib/Stim/blob/main/doc/gates.md
+        qiskit_to_stim_dict = {
+            "id": "I",
+            "x": "X",
+            "y": "Y",
+            "z": "Z",
+            "h": "H",
+            "s": "S",
+            "sdg": "S_DAG",
+            "cx": "CNOT",
+            "cy": "CY",
+            "cz": "CZ",
+            "swap": "SWAP",
+            "reset": "R",
+            "measure": "M",
+            "barrier": "TICK",
+        }
+        pauli_error_1_stim_order = {"id": 0, "I": 0, "X": 1, "x": 1, "Y": 2, "y": 2, "Z": 3, "z": 3}
+        pauli_error_2_stim_order = {
+            "II": 0,
+            "IX": 1,
+            "IY": 2,
+            "IZ": 3,
+            "XI": 4,
+            "XX": 5,
+            "XY": 6,
+            "XZ": 7,
+            "YI": 8,
+            "YX": 9,
+            "YY": 10,
+            "YZ": 11,
+            "ZI": 12,
+            "ZX": 13,
+            "ZY": 14,
+            "ZZ": 15,
+        }
+
+        measurement_data = []
+        register_offset = {}
+        previous_offset = 0
+        for inst, qargs, cargs in circuit.data:
+            for qubit in qargs:
+                if qubit._register.name not in register_offset:
+                    register_offset[qubit._register.name] = previous_offset
+                    previous_offset += qubit._register.size
+
+            qubit_indices = [
+                qargs[i]._index + register_offset[qargs[i]._register.name]
+                for i in range(len(qargs))
+            ]
+
+            if isinstance(inst, QuantumChannelInstruction):
+                # Errors: it would be good to add exeptions for delpoarizinng noise, Z error etc,
```

Comment:
Can you remove this comment if you've already made sure non-Clifford gates throw exceptions? Then, you can make a new issue for non-clifford gate errors specifically?